test/grpc_test.rb

```diff
@@ -190,7 +191,7 @@ def build_rpc_server(server_opts: {}, client_opts: {})
     @server = new_rpc_server_for_testing({ poll_period: 1 }.merge(server_opts))
     @port = @server.add_http2_port("0.0.0.0:#{SemianConfig["grpc_port"]}", :this_port_is_insecure)
     @host = "#{@hostname}:#{@port}"
-    @client_opts = client_opts
+    @client_opts = { timeout: DEFAULT_CLIENT_TIMEOUT_IN_SECONDS }.merge(client_opts)
```

Comment:
though gRPC code says default is infinity, but my local test also align with CI, seems that 20s is set
https://github.com/grpc/grpc/blob/53d69cc581c5b7305708587f4f1939278477c28a/src/ruby/lib/grpc/generic/client_stub.rb#L28-L29

20s is somewhat same as the example, maybe an coincide
https://grpc.io/blog/deadlines/#adjusting-deadlines